lib/stat_tracker.rb

```diff
@@ -493,4 +491,51 @@ def worst_defense
     team_id = all_teams_opponent_averages.key(all_teams_opponent_averages.values.max)
     get_team_name_from_id(team_id)
   end
+
+  def find_games_by_team_id(team_id, games = @games)
+    games.select do |game|
+      game.away_team_id == team_id || game.home_team_id == team_id
+    end
+
+  end
+
+  def group_games_by_season_and_team(season_id, team_id)
+    all_games_in_season = games_by_season[season_id]
+    all_games_a_team_played = find_games_by_team_id(team_id)
+    all_games_in_season & all_games_a_team_played
+  end
+
+  def season_summary(season_id, team_id)
+    team = find_team(team_id)
+
+    games = group_games_by_season_and_team(season_id, team_id)
+    preseason = group_games_by_season_type("P", games)
+    reg_season = group_games_by_season_type("R", games)
+
+    preseason_hash = { :win_percentage => team.win_percentage(preseason),
+                        :goals_scored => team.goals_scored(preseason),
+                        :goals_against => get_opponent_goals(team_id, preseason)}
+
+    reg_season_hash = { :win_percentage => team.win_percentage(reg_season),
+                        :goals_scored => team.goals_scored(reg_season),
+                        :goals_against => get_opponent_goals(team_id, reg_season)}
+
+    {:preseason => preseason_hash,
+    :regular_season => reg_season_hash}
+  end
+
+  def get_opponent_goals(team_id, games = @games)
```

Comment:
here, I wanted to use the team_opponent_goals from the game_team instance method. Problem was, I only wanted it for certain games (preseason and then reg season ones.